cmd/ro/main.go

```diff
@@ -207,6 +227,85 @@ func runDecrypt() {
 	ioutil.WriteFile(outPath, msg.Data, 0644)
 }
 
+func runSSHAgent() {
+	inBytes, err := ioutil.ReadFile(inPath)
+	processError(err)
+
+	// base64 decode the input
+	encBytes, err := base64.StdEncoding.DecodeString(string(inBytes))
+	if err != nil {
+		log.Println("failed to base64 decode the data, proceeding with raw data")
+		encBytes = inBytes
+	}
+
+	inBytes, err = ioutil.ReadFile(pubKeyPath)
+	processError(err)
+
+	pubKey, _, _, _, err := ssh.ParseAuthorizedKey(inBytes)
+
+	if err != nil {
+		log.Fatal("failed to parse SSH public key", err)
+	}
+
+	roagent := roagent.NewROAgent(roServer, pubKey, encBytes, user, pswd)
+
+	authSockPath := os.Getenv("SSH_AUTH_SOCK")
+	if authSockPath == "" {
+		log.Fatal("SSH_AUTH_SOCK not set")
+	}
+
+	listener, err := net.ListenUnix("unix", &net.UnixAddr{Name: authSockPath, Net: "unix"})
+	if err != nil {
+		log.Fatal("error listening on $SSH_AUTH_SOCK", err)
+	}
+
+	defer os.Remove(authSockPath)
+
+	conn, err := listener.AcceptUnix()
+	if err != nil {
+		log.Fatal("error accepting socket connection", err)
+	}
+
+	err = agent.ServeAgent(roagent, conn)
+	if err != nil && err != io.EOF {
+		log.Fatal("error serving socket protocol", err)
+	}
+}
+
+func runSSH() {
+	// First pick a path for our socket
+	// TempDir will ensure that the directory is created with the correct permissions
+	dir, err := ioutil.TempDir("", "ro_ssh_")
+	if err != nil {
+		log.Fatal("error getting temporary directory for SSH auth socket ", err)
+	}
+	defer os.RemoveAll(dir)
+
+	os.Setenv("SSH_AUTH_SOCK", path.Join(dir, "roagent.sock"))
+	go runSSHAgent()
+
+	var sshPath string
+	sshPath, err = exec.LookPath("ssh")
+	if err != nil {
+		log.Fatal("error finding path to ssh binary ", err)
+	}
+
+	var p *os.Process
+	p, err = os.StartProcess(sshPath, flag.Args(),
+		&os.ProcAttr{
+			Files: []*os.File{os.Stdin, os.Stdout, os.Stderr},
+		},
+	)
+	if err != nil {
+		log.Fatal("error starting ssh ", err)
+	}
+
+	_, err = p.Wait()
```

Comment:
Think there would ever be a case where it would useful to report on exit code not being 0?